### DISCUSSION TITLE
GraphQL sample code via plain HTTP posts missing

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#2488] [Feature Requests]
Supabase advertises with "Query your tables, including filtering, pagination, and deeply nested relationships (like GraphQL)". Is this only possible via supabase-js (and other clients), or also via **plain http payloads**?

In general it was already hard to find info on how to query  PostgREST as nowhever was mentioned you need to add "/rest/v1/" between your hostname and the https://postgrest.org/en/stable/api.html manual.

**Comment by steve-chavez** [comment]
On the API docs(`<your_project_ref>/api/default`) there are Bash snippets that show you plain `curl`(http) calls.

**Reply by ghost** [reply]
I'm familiar with those. They are also shown on https://postgrest.org/en/stable/api.html, but nevertheless I should have seen these here as all details are mentioned.

There's however no sample advanced query with a table and junction table like:

`
const { data, error } = await supabase
  .from('users')
  .select(`
    id,
    user,
    userProducts:products(
      id,
      name,
      price,
      description,
      tags(
        id,
        name
      )
    )
 `)
`

Any suggestion?

**Comment by ghost** [comment]
Additionally I'm missing the basic Storage samples. How to upload and download a file using bash. Any suggestions?
